### PR TITLE
First stab at fpaste.org (Issue #2)

### PR DESCRIPTION
**Pull request by aggrolite**
Hi Joel,

This change adds support for fpaste.org. Should solve issue #2. Let me know if there are any problems - feedback always welcome.

### CODE REVIEW

File: lib/Mojolicious/Command/nopaste/Service/fpaste.pm [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package Mojolicious::Command::nopaste::Service::fpaste;
+use Mojo::Base 'Mojolicious::Command::nopaste::Service';
+
+has description => "Post to fpaste.org\n";
+
+sub paste {
+  my $self = shift;
+  my $lang = $self->language || 'text';
```
Comment (jberger) [inline lib/Mojolicious/Command/nopaste/Service/fpaste.pm]: most (all?) of the services default to perl syntax so far. this is of debatable value, but until a change is made, perhaps aim for consistency?

File: lib/Mojolicious/Command/nopaste/Service/fpaste.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  unless ($tx->res->is_status_class(200)) {
+    say $tx->res->message;
+    say $tx->res->body;
+    exit 1;
+  }
+
+  my $url = 'http://fpaste.org/';
+
+  $url .= join '/', @{$tx->res->json->{result}}{qw(id hash)};
```
Comment (jberger) [inline lib/Mojolicious/Command/nopaste/Service/fpaste.pm]: why not use Mojo::URL object and push onto the path parts?

File: lib/Mojolicious/Command/nopaste/Service/fpaste.pm (hunk 1 above)
Comment (aggrolite) [inline lib/Mojolicious/Command/nopaste/Service/fpaste.pm]: I was going by what I read in `debian.pm` and `pastie.pm` which use text if `$self->language` does not return a true value. Just pointing that out - but I will happily change to `my $lang = $self->language || 'perl'` if desired.

File: lib/Mojolicious/Command/nopaste/Service/fpaste.pm (hunk 2 above)
Comment (aggrolite) [inline lib/Mojolicious/Command/nopaste/Service/fpaste.pm]: Good idea - thanks. :smile:

**Comment by aggrolite** [comment]
New commit pushed for review.

File: lib/Mojolicious/Command/nopaste/Service/fpaste.pm (hunk 1 above)
Comment (jberger) [inline lib/Mojolicious/Command/nopaste/Service/fpaste.pm]: Ah, indeed, you are right. The default of "perl" comes from here https://github.com/jberger/Mojolicious-Command-nopaste/blob/master/lib/Mojolicious/Command/nopaste/Service.pm#L50 and then if it gets to the individual service without a value, then it should be plain text. Could you please change that one back?

Comment (aggrolite) [inline lib/Mojolicious/Command/nopaste/Service/fpaste.pm]: Fixed here https://github.com/aggrolite/Mojolicious-Command-nopaste/commit/e0c721ea81102a672f82290a2331df3341d27e9a

**Comment by aggrolite** [comment]
Just FYI - `slurp()` appears to add on an extra empty line to file content.

For example, http://fpaste.org/232342/ shows 3 lines (the last one being empty), when the file I pasted was only 2 lines:

```perl
use Moose;
1;
```

**Comment by jberger** [comment]
I could argue that that is a feature, but yeah, I'm not seeing where that happens. I'm at $work currently so I can't look closely (yet)

**Comment by aggrolite** [comment]
No problem - I can write up something more detailed and file as a separate issue.

**Comment by jberger** [comment]
Yes that would certainly be a separate issue and we can then discuss there. Thanks.

**Comment by aggrolite** [comment]
merging this change. Separate issue regarding slurp filed here: https://github.com/jberger/Mojolicious-Command-nopaste/issues/4

**Pull request merged**